src/CoreWCF.BuildTools/src/OperationParameterInjectionGenerator.Emitter.cs

```diff
@@ -0,0 +1,216 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CoreWCF.BuildTools
+{
+    public sealed partial class OperationParameterInjectionGenerator
+    {
+        private sealed class Emitter
+        {
+            private class Indentor
+            {
+                const string ____ = "    ";
+                const string ________ = "        ";
+                const string ____________ = "            ";
+                const string ________________ = "                ";
+                const string ____________________ = "                    ";
+                const string ________________________ = "                        ";
+                const string ____________________________ = "                            ";
+                const string ________________________________ = "                                ";
+                public int Level { get; private set; } = 0;
+                public void Increment()
+                {
+                    Level++;
+                }
+
+                public void Decrement()
+                {
+                    Level--;
+                }
+
+                public override string ToString() => Level switch
+                {
+                    0 => string.Empty,
+                    1 => ____,
+                    2 => ________,
+                    3 => ____________,
+                    4 => ________________,
+                    5 => ____________________,
+                    6 => ________________________,
+                    7 => ____________________________,
+                    8 => ________________________________,
+                    _ => throw new InvalidOperationException(),
+                };
+            }
+
+            private readonly OperationParameterInjectionSourceGenerationContext _sourceGenerationContext;
+            private SourceGenerationSpec _generationSpec;
+
+            public Emitter(in OperationParameterInjectionSourceGenerationContext sourceGenerationContext, SourceGenerationSpec generationSpec)
+            {
+                _sourceGenerationContext = sourceGenerationContext;
+                _generationSpec = generationSpec;
+            }
+
+            public void Emit()
+            {
+                foreach (var operationContractSpec in _generationSpec.OperationContractSpecs)
+                {
```

Comment:
You've enabled Nullable C# feature in the csproj (which is great BTW), but there are still a small number of nullability issues in the code. For example, there are 5 instances of "Warning CS8602 Dereference of a possibly null reference." I know there are a lot of warnings throughout the codebase that need to be addressed, but I plan to get it to a clean build where we can turn on treat warnings as errors.

Reply:
Done here https://github.com/CoreWCF/CoreWCF/pull/493/commits/c150ef935aba68d117b05cdf2daefd08e5bb1f48.